geps/gep-1619.md

```diff
@@ -157,49 +281,401 @@ Generally, the implementation API programs the dataplane API; however these two
+
+##### Path
+
+The `Path` attribute should be automatically determined by implementors. When the `SessionPersistencePolicy` policy is
+attached to an xRoute, the implementor should interpret the path as configured on the xRoute. However, if the
+`SessionPersistencePolicy` policy is attached to a service, the path should be left unset.
+
+To interpret the `Path` attribute from an xRoute, implementors should take note of the following:
+1. For an xRoute that matches all paths, the `Path` should be set to `/`.
+2. For an xRoute that has multiple paths, the `Path` should be interpreted based on the path taken.
+3. For an xRoute using a path that is a regex, the `Path` should be set to the longest non-regex prefix (.e.g. if the
+   path is /p1/p2/*/p3 and the request path was /p1/p2/foo/p3, then the cookie path would be /p1/p2).
+
+##### Secure, HttpOnly, SameSite
+
+The `Secure`, `HttpOnly`, and `SameSite` cookie attributes are security-related. The API implementers should follow the
+security-by-default principle and configure these attributes accordingly. This means enabling `Secure` and `HttpOnly`,
+and setting `SameSite` to `Strict`.
+
+#### Session Persistence API with GAMMA
+
+The object of the [GAMMA (Gateway API for Mesh Management and Administration)](https://gateway-api.sigs.k8s.io/contributing/gamma/)
+initiative is to provide support for service mesh and mesh-adjacent use-cases with Gateway API. Mesh traffic routing with
+Mesh traffic routing with Gateway API is accomplished with `xRoutes` specifying service-to-service traffic opposed to
+gateway-to-service traffic.
+
+Given that service meshes commonly have session persistence requirements, this API design should take into consideration
+session persistence needs in GAMMA and service mesh scenarios.
+
+#### Session Initiation Guidelines
+
+As illustrated in the examples provided in [Session Persistence Initiation](#session-persistence-initiation),
+implementations must consider how to manage sessions initiated by other components. This GEP leaves the decision of
+managing existing sessions with each specific implementation. In the case of cookie-based session persistence, an
+implementation has the freedom to either rewrite the cookie or insert an additional cookie, or to do nothing (resulting
+in the lack of a persistent session).
+
+#### Expected API Behavior
+
+Implementing session persistence is complex and involves many edge cases. In this section, we will outline API
+configuration scenarios (use cases) and how implementations should handle them.
+
+##### Attaching Session Persistence to both Service And Route
+
+In a situation which:
+- `ServiceA` with `SessionPersistencePolicy` attached
+- `RouteX` with `SessionPersistencePolicy` attached and backend `ServiceA`
+
+The `SessionPersistencePolicy` policy attached to `RouteX` should take precedence. Since routes effectively group
```

Comment:
Not sure what this means. It's not authz - with DENY and ALLOW. 

If a service is designed for persistence, and user indicates it using a Service attachment - it should be respected. If it also defines it in route - good. If a route doesn't indicate persistence but uses that backend - it should still respect the user intent. 
 
The only case where a Route would take precendence is if Service doesn't define persistence - i.e. migration cases where v1 doesn't support persistence and v2 does ( or reverse). Unless v1 is able to deal 
with persistence ( mainly different load distribution ) it may be dangerous to do that. 

Not a huge issue maybe (I don't think it'll be that common for users to have both), but we should be clear with the risks.

Reply:
Right. My intention with this "precedence" scenario is that if the specific configuration for session persistence is different between the Service and the Route that is targeting this service. I.e, what if you had a different `durationSeconds` on the attachment to the service and the route? Which one takes for the persistence?

I added a sentence in `API Attachment Point: Route vs. Service` to make sure it was clear you don't need both enabled.

Reply:
SGTM.